### PR TITLE
func_801B2FD8 Matching

### PR DESCRIPTION
**Pull request by sozud**
@sonicdcer is asking me to put up some PRs for him, here's the first.

### CODE REVIEW

File: src/st/nz0/30958.c [hunk 1]
```diff
@@ -898,7 +898,85 @@ void func_801B2AD8(Entity* self) {
...
+        posX = self->posX.i.hi;
+        posX += g_Camera.posX.i.hi;
+        posX >>= 4;
+
+        // TODO: !FAKE
+        camY = self->posY.i.hi;
+        posY = camY += 4;
+        posY = camY += g_Camera.posY.i.hi;
+
+        g_CurrentRoomTileLayout
```
Comment (sozud) [inline src/st/nz0/30958.c:940]: @sonicdcer I think this is probably more likely
```
            camY = (camY >> 4) * g_CurrentRoom.hSize * 16;
            g_CurrentRoomTileLayout.fg[posX + camY] = 0x5AF;
```
Other than that LGTM.

Comment (sonicdcer) [reply src/st/nz0/30958.c:940]: If it matches, yeah go ahead :)

Comment (sozud) [reply src/st/nz0/30958.c:940]: It does https://decomp.me/scratch/cB84r
This PR is from your made fork so if you agree with the change can you push it?

Comment (sonicdcer) [reply src/st/nz0/30958.c]: Done!

**Comment by sozud** [comment]
This one also looks pretty straightforward to me so merging to reduce the queue.

**Pull request merged**